collections_hierarchy/mixins.py

```diff
@@ -1,87 +1,103 @@
 class ComparableMixin(object):
     def __eq__(self, other):
-        pass
+        for x, y in zip(self, other):
+            if not x == y:
+                pass
+            else:
+                return self == other
 
     def __ne__(self, other):
-        return not self == other
+        for x, y in zip(self, other):
+            if x == y:
+                pass
+            else:
+                return not self == other
 
     def __lt__(self, other):
-        pass
+        return self < other
 
     def __gt__(self, other):
-        pass
+        return self > other
 
     def __le__(self, other):
-        pass
+        return self <= other
 
     def __ge__(self, other):
-        pass
+        return self >= other
 
 
 class SequenceMixin(object):
     def __iter__(self):
-        pass
+        self.n = 0
```

Comment:
I'm not sure if you can define new class attributes in `__iter__()`, but the logic behind this together with `__next__()` makes sense.  I haven't had a chance to run the code yet, but does this work to make your class instances iterable?  If not, is there some way you can set all instance attributes in `__init__()` and then just use them here?